public/app/features/explore/QueryLibrary/types.ts

```diff
@@ -1,11 +1,3 @@
-import { ComponentType } from 'react';
-
 import { DataQuery } from '@grafana/schema';
 
-export type QueryActionButtonProps = {
-  queries: DataQuery[];
-  datasourceUid?: string;
-  onClick: () => void;
-};
-
-export type QueryActionButton = ComponentType<QueryActionButtonProps>;
+export type OnSelectQueryType = (query: DataQuery) => void;
```

Comment:
I don't think this will be enough: https://github.com/grafana/grafana/blob/0029f92ac3a5c0d1bf2a24ec3d99c0dff97653aa/packages/grafana-schema/src/common/common.gen.ts#L40

Not really sure now if that comment is 100% accurate but at least it seems the datasource prop is not a guarantee.
That would mean we won't know if user selected the same datesource or some other. If we decide not to allow selecting different data source then it may be moot but not sure if we locked ourselves to that already or not.

Reply:
i think we can assume it will always be there. it already does when we save a query to the library.

Reply:
investigated this more, and confirmed that any query in the query library will always have a datasource 👍

Reply:
Ah ok didn't realize we recreate the DataQuery when saving.